implementation/src/main/java/io/smallrye/mutiny/operators/uni/UniOperatorProcessor.java

```diff
@@ -13,15 +13,19 @@
 public abstract class UniOperatorProcessor<I, O> implements UniSubscriber<I>, UniSubscription {
 
     protected final UniSubscriber<? super O> downstream;
-    protected final AtomicReference<UniSubscription> upstream = new AtomicReference<>();
+
+    private static final AtomicReferenceFieldUpdater<UniOperatorProcessor, UniSubscription> updater = AtomicReferenceFieldUpdater
+            .newUpdater(UniOperatorProcessor.class, UniSubscription.class, "upstream");
+
+    private volatile UniSubscription upstream;
 
     public UniOperatorProcessor(UniSubscriber<? super O> downstream) {
         this.downstream = ParameterValidation.nonNull(downstream, "downstream");
     }
 
     @Override
     public void onSubscribe(UniSubscription subscription) {
-        if (upstream.compareAndSet(null, subscription)) {
+        if (updater.compareAndSet(this, null, subscription)) {
```

Comment:
Sorry, but why not reuse the `compareAndSetUpstreamSubscription` method described below (line 74)?

Reply:
That's right thanks!